### PR TITLE
feat(application): add async lifecycle coordination

### PR DESCRIPTION
**Pull request by paulfalgout**
Related #131
Related #190

## What

- add Promise-based `start`, `stop`, `restart`, and Application-specific `destroy` lifecycle coordination
- add deterministic supersession, compatible in-flight sharing, terminal destruction, and `isRunning()`
- document and test async readiness, rejection, reentry, overlap, Radio teardown, repeated cycles, and migration from v4

## Why

Application needs one explicit readiness and shutdown contract before a useful v5 alpha can be exercised in asynchronous consumers. V3/v4 synchronous `start()` is historical evidence, but copying it would leave readiness implicit and would not define stop, restart, or overlapping cleanup.

## Agent-development failure addressed

The prior implicit lifecycle required consumers and coding agents to infer when startup was complete and how overlapping startup, shutdown, and destruction should settle. The new return values distinguish reached state (`true`), ordinary supersession (`false`), and current hook failure (rejection).

## Public behavior contract

- Canonical behavior after this PR: Application lifecycle methods return `Promise<boolean>`; compatible calls share the active Promise; the latest incompatible live operation wins; destroy is terminal.
- Superseded behavior removed: Application `start()` no longer synchronously returns the Application, and Toolkit-style parallel start-finalization hooks are not added to core.
- Compatibility exception: none. Consumers must await lifecycle calls when subsequent work requires the requested state.

## Scope

This changes the production Application lifecycle and generated root distributables. It intentionally does not add child Applications, Region ownership changes, topology queries, State, Toolkit compatibility aliases, or fallback lifecycle paths. The optional Application Region behaves as before.

## Runtime cost

- [ ] Static or documentation only
- [ ] Development or test only; excluded from production entrypoints
- [x] Changes an existing production path
- [ ] Adds an explicitly opt-in runtime path

Measured impact or explanation:

- Bundle: 67.51 kB cumulative Brotli-11 against the 75.00 kB ceiling; exact-base growth exceeds 1% and requires the repository's exact-head approval record.
- Startup/hot path: lifecycle coordination runs only when an Application lifecycle method is called; hosted timing remains reporting-only.
- Allocations/retention: one in-flight operation record and Promise readiness state while a lifecycle transition is active; no record remains after settled repeated cycles.

## Validation

- [x] Tests cover the acceptance criteria and relevant edge cases.
- [x] Docs, examples, declarations, diagnostics, and rule metadata agree where relevant.
- [x] No private consumer, private fixture, or undocumented context is required.
- [x] I ran the commands listed below and am reporting their actual results.

Commands:

```sh
npm test -- --run test/unit/application-lifecycle.spec.js --reporter=dot # 35 passed
npm run coverage -- --reporter=dot # 95 files, 1775 tests, 100% statements/branches/functions/lines
npm run lint:ci # passed
npm run docs:check # 43 HTML files, 394 internal links
npm run check:dist # passed on committed generated artifacts
npm run test:source # passed
npm run test:fixtures # passed, including packed CJS/ESM/types/Vite consumers
npm run test:performance # 105 passed
npm run size # 67.51 kB cumulative / 75.00 kB
npm run performance:timing # completed; reporting only
git diff --check # passed
```

Edge cases include stop/restart/destroy supersession before and after the restart stop phase, transitive shared stop-readiness failure, synchronous hook failure, completion-handler reentry, four-call overlap, stale rejection absorption, Radio reply teardown, terminal no-ops, and repeated start/stop cycles.

## Package and browser impact

- Entry points or install fixtures affected: root `marionette` runtime and its generated ESM, CommonJS, UMD, and minified UMD artifacts; no new subpath.
- Browser contracts affected: Application lifecycle timing and return values; the DOM and Region contracts are unchanged.
- Production/dev/test module-graph impact: no production module-graph expansion; generated root artifacts grow and require exact-head approval.

## Deployment Impact

No website, npm package, tag, release, form bundle, or deployment is created by this PR. Merging changes repository source only. Future consumers need a later package build and must await lifecycle operations where readiness matters.

## Rollback or deprecation

Before stable v5, rollback is a source revert of this PR. No compatibility alias or dual synchronous lifecycle path is retained.

## Review notes

- Exact-head Claude review is required independently of CI and automated reviewers.
- Review feedback found and fixed duplicate restart teardown plus incorrect failure-state inheritance across stop/restart/destroy supersession.
- Bundle growth approval must target the final exact head and all required artifact paths.

## Summary by CodeRabbit

* **New Features**
  * Application lifecycle operations are now asynchronous and return promises.
  * Added `stop`, `restart`, and `isRunning` lifecycle capabilities.
  * Lifecycle readiness hooks support cooperative cancellation through abort signals.
  * Later lifecycle actions safely supersede incompatible in-progress operations.
* **Documentation**
  * Updated lifecycle, installation, event, and migration guidance with the new asynchronous behavior.
  * Added examples for awaiting lifecycle operations and handling cancellation.

### CODE REVIEW

**Comment by paulfalgout** [comment]
**Not approving yet** at `4d21f9fd` (the SHA I fetched). Three checks are red with one root cause, and the design underneath is sound — this is a coverage gap, not a defect.

## Node 24 and Build verified tarball both fail on `Run configured coverage`

Lines 99.89% and branches 99.38% against a 100% gate; `application.js` alone is 95.27% statements / 91.46% branches. Two lines are named, and they are worth covering rather than being coverage padding — both are the error paths where async lifecycle code actually goes wrong:

**`application.js:134`** — the `reject(error)` inside

```js
try {
  Promise.resolve(application.triggerMethod('before:stop', application, options))
    .then(resolve, reject);
} catch (error) {
  reject(error);
}
```

`Promise.resolve(...).then(resolve, reject)` handles a handler that returns a rejected promise. It does **not** handle a handler that throws **synchronously** before returning anything — that unwinds past `Promise.resolve` and into the `catch`. Distinguishing those two is the entire reason the `try` is there, and nothing exercises it. A synchronous `throw` in a `before:stop` listener is not exotic.

**`application.js:189`** — the `return true` in

```js
return operation.promise.catch(error => {
  if (this._lifecycleState === STOPPED || this._lifecycleState === DESTROYED) {
    return true;
  }
  throw error;
});
```

This is `stop()` called during `DESTROYING` where the destroy operation rejects, but the application nonetheless reached STOPPED or DESTROYED — so the stop succeeded even though the destroy did not. That single line encodes "these two operations can have different outcomes," which is a real contract claim, and it is the one untested branch of it.

## Bundle size, secondary but worth naming

**+3,155 B, +4.91% aggregate** (67,379 B against the 75,000 B ceiling). All four main artifacts are 4.72%–5.07%, so a maintainer growth approval is required on top of the coverage fix.

Headroom is now **7,621 B**, down from 10,776 B one PR ago and ~13.9 kB a dozen PRs back. No single change has been unreasonable, but this is the largest step so far and the direction has been consistent. Worth knowing the intended landing zone before the remaining #131/#132 slices rather than after.

## The design is right, which is why the blockers read as mechanical

- **Supersession resolves rather than rejects.** `supersedeOperation` settles the displaced promise with `!!operation.completing` and never rejects it, so a superseded operation cannot produce a floating unhandled rejection. That is the single most consequential decision in this file and it is made correctly.
- **`isCurrentOperation` guards both `completeOperation` and `failOperation`**, so a stale operation cannot clobber `_lifecycleState` or reject after being displaced.
- **`completing` is set after the state transition and before the final `triggerMethod`.** That makes the resolved boolean meaningful: `true` means the operation reached its goal and was superseded during the terminal event, `false` means it was superseded before arriving. That is a genuine contract rather than an artifact.
- **The `stopReadiness` handoff is the subtle part and it is right.** A start that supersedes a stop awaits the *same* `before:stop` settlement instead of re-firing it, because `before:stop` is not safely repeatable — while `before:start` can be re-run with the new call's options. The asymmetry between how the two hooks are handled is deliberate and correct.

## One thing to keep prominent

`Application#destroy` now returns a Promise, overriding `DestroyMixin`'s `return this`. Chaining off `app.destroy()` breaks for Application specifically while continuing to work for every other class using that mixin. The migration ledger and `mixins/destroy.spec.js` have both been updated, so this is handled — but a divergence where one class's `destroy` returns something different from its siblings is the kind of thing worth stating in the class reference too, not only the migration table.

Happy to re-review once coverage is closed.

**Comment by paulfalgout** [comment]
**Approved on the code** at `ee308a0d`. Both named coverage gaps now have tests, and the branch-coverage half was closed by deleting unreachable code rather than contriving a test for it — which is the honest resolution. CI was still re-running when I looked, so the gate itself needs to confirm; **Bundle size will still need a growth approval** at this head (+3,252 B, +5.06%, 67,476 B against the 75,000 B ceiling, 7,524 B headroom).

## Removing the two `isCurrentOperation` guards in `destroy` is correct

I checked whether an in-flight destroy can be superseded at all, since those guards existing implied it could. It cannot — every public entry point refuses before reaching `beginOperation`:

- `start()` — returns `false` during `DESTROYING`/`DESTROYED`
- `restart()` — returns `false` during `DESTROYING`/`DESTROYED`
- `stop()` — returns the destroy's own promise during `DESTROYING`
- `destroy()` — joins the existing destroy operation

Since `beginOperation` sets `_lifecycleState = DESTROYING` before running the callback, nothing can displace `_lifecycleOperation` afterward. Those two branches were unreachable, and deleting them is both correct and the right way to satisfy a branch-coverage gate.

The asymmetry that remains is exactly right: `restart` **keeps** its guard after `stopApplication`, because restart genuinely is interruptible — `start`, `stop`, and `destroy` all proceed to `beginOperation` during `RESTARTING`. Destroy is terminal and needs no guard; restart is not and keeps one. That reads as deliberate rather than inconsistent.

## The `completing` / `stopped` split corrects something I got wrong

Last review I described `completing` as "a genuine contract rather than an artifact." It was, for `stop` — and subtly wrong for the composite operations. `stopApplication` set `completing = true` unconditionally, so a **restart** or **destroy** superseded during its stop phase would have resolved `true`, claiming it reached its goal when it had only finished a sub-phase.

Splitting them fixes that: `completing` now means "this operation achieved its own kind" and is set only when `kind === 'stop'`, while `stopped` separately records that the stop phase finished. `shouldStop` keying off `stopped` rather than the overloaded `completing` follows from the same correction. I praised the old flag without noticing it was doing two jobs.

## On the new tests

Writing **two** cases for `application.js:189` — one per disjunct of `STOPPED || DESTROYED` — is the right granularity for a branch gate rather than one test that happens to hit whichever branch runs first. And `rejects a synchronous before:stop failure` targets exactly the distinction that made the `try` necessary: a handler that throws before returning, which `Promise.resolve(...).then(resolve, reject)` cannot catch.

**Comment by paulfalgout** [comment]
**Approval stands** at `90c3fdfe`. One targeted fix, and it closes a real asymmetry rather than a cosmetic one.

```diff
-    const failureState = this._lifecycleState === RUNNING ? RUNNING : STOPPED;
+    const failureState = operation?.stopReadiness ? operation.failureState :
+      this._lifecycleState === RUNNING ? RUNNING : STOPPED;
```

`start()` already inherited `failureState` from a superseded operation carrying `stopReadiness`; `stop()` did not. That gap mattered because `beginOperation` hands the shared readiness to the winning operation, so the two paths were resolving the same shared failure to different states.

The concrete case is the one the new test names. A restart mid-teardown holds `failureState === RUNNING`, since RUNNING is what it must fall back to. If `stop()` supersedes it and the **shared** `before:stop` then rejects, the old expression read `_lifecycleState` — which is `RESTARTING`, not `RUNNING` — and settled on `STOPPED`. So a stop that explicitly failed would have left the Application reporting itself stopped, having never stopped. Inheriting the superseded operation's `failureState` keeps it `RUNNING`, which is the truth.

Worth noting the failure is silent in the wrong direction: a rejected promise the caller might reasonably ignore, paired with state that claims success. Pairing it with `synchronously` in the test name is deliberate too — that is the `application.js:134` path from earlier, so the two fixes now intersect in one case rather than being covered only in isolation.

The growth approval target moves to this head; sizes are unchanged from my last comment in any material way.

**Comment by paulfalgout** [comment]
**Approval stands** at `8abf121b`. Extracting `getFailureState` reads like deduplication but is a behavior change at three of the four call sites, and the right one.

Before this push, only `start` and `stop` consulted `stopReadiness`, and they disagreed on the fallback — `start` returned `STOPPED` unconditionally, `stop` derived it from `_lifecycleState`. `restart` and `destroy` consulted neither. Four entry points, three different answers to the same question.

Now all four answer it identically: inherit the superseded operation's `failureState` when a shared stop readiness is in flight, otherwise fall back to whether the Application is currently running. The three new tests target exactly the sites that previously got it wrong — `restart` inheriting failing stop readiness, `destroy` inheriting it, and `destroy` superseding a failing restart teardown — rather than re-testing the `stop` case that was already fixed.

Worth noting the `start` fallback changed too, and it is not incidental. It was hardcoded `STOPPED`; it now yields `RUNNING` when the Application is running. That path is reachable: `start()` returns early on RUNNING only when there is **no** in-flight operation, so a `start` listener calling `start()` again during `triggerMethod('start')` proceeds to `beginOperation` with state already RUNNING. If that nested start failed, the old code marked a running Application stopped.

This is the fourth push refining `failureState`, and the sequence has converged rather than oscillated — found in `stop`, generalized to the family, with each step keeping the prior tests. One helper is also easier to keep honest than four expressions that already drifted apart once.

Growth approval target moves to this head: +3,289 B, +5.12%, 7,487 B headroom.

**Comment by paulfalgout** [comment]
Claude exact-head review for `8abf121b9ec539e3a2dbd334ed8d9bcace67d808`: **APPROVE — no blocking issues**.

Claude reviewed the complete Application operation helpers and public lifecycle methods, then manually traced start/stop/restart/destroy interleavings, shared `stopReadiness`, failure-state propagation, stale continuations, and terminal destroy behavior. Earlier review rounds found two real defects: restart stop completion could be repeated or report stale success, and restart/destroy could lose the original RUNNING failure state when inheriting a pending stop. The exact head records completed stop phases, centralizes failure-state inheritance, and covers stop→restart, stop→destroy, restart→destroy, restart→stop, repeated stop→destroy, synchronous rejection, and terminal no-op sequences.

Local exact-tree validation: 35 focused lifecycle tests; 1,775 full tests with 100% statements, branches, functions, and lines; lint, docs, generated distributions, source validation, packed fixtures, performance contract tests, size, and timing completed successfully. GitHub CI, Cubic, exact-head growth approval, mergeability, and unresolved threads remain independent gates.

File: modules/application.js [hunk 1]
```diff
@@ -28,17 +36,213 @@ const Application = function(options) {
...
+
+      return operation.promise.catch(error => {
+        if (this._lifecycleState === STOPPED || this._lifecycleState === DESTROYED) {
+          return true;
+        }
+        throw error;
+      });
+    }
+    if (operation?.kind === 'stop') { return operation.promise; }
+    if (this._lifecycleState === STOPPED && !operation) { return Promise.resolve(true); }
```
[the remark this answers is not in the document]
Comment (paulfalgout) [reply modules/application.js:248]: Handled in modules/application.js and the restart supersession regression tests; a completed stop phase is reused and the invalidated restart resolves false.

File: docs/events.class.md [hunk 2]
```diff
@@ -81,13 +81,13 @@ const MyApp = Application.extend({
 });
 
 const myApp = new MyApp({ foo: 'My App' });
-myApp.start({ data: { bar: true } });
+await myApp.start({ data: { bar: true } });
```
[the remark this answers is not in the document]
Comment (paulfalgout) [reply docs/events.class.md:86]: Handled in docs/events.class.md with before:stop and stop reference entries and the corrected v5 package import.

**Comment by paulfalgout** [comment]
**Approval stands** at `389ef16c`. This is the best version of the `stop()`-during-`DESTROYING` path so far, and it retires a line I flagged in my first review by making it unnecessary rather than by testing it harder.

```diff
-      return operation.promise.catch(error => {
-        if (this._lifecycleState === STOPPED || this._lifecycleState === DESTROYED) {
-          return true;
-        }
-        throw error;
-      });
+      operation.stopPromise ||= operation.stopReadiness.promise.then(() => true);
+      return operation.stopPromise;
```

The old form awaited the whole **destroy** promise and then inferred whether the stop had succeeded by inspecting `_lifecycleState`. The new form awaits the **stop readiness** directly. That is the right question: `stop()` should report on the stop phase, not on whatever the destroy does afterward. If `before:stop` succeeds and `before:destroy` then fails, the stop genuinely succeeded — previously that had to be reconstructed from state, now it falls out.

Rejection still propagates correctly, since `.then(() => true)` has no rejection handler, so a failing `before:stop` rejects the caller's `stop()`.

I originally called `application.js:189` "the one line encoding that these two operations can have different outcomes." The answer turned out to be that no line needed to encode it — awaiting the correct promise makes the distinction structural. Both earlier tests for that behavior are still present and still meaningful, which is the check I wanted: the contract is unchanged, only its derivation is.

`||=` memoizing `stopPromise` matters for the repeat case rather than being a micro-optimization — several `stop()` calls during one destroy now share a single settlement instead of each building a separate `.then` chain over the same readiness. `shares stop calls during destroy and settles after the stop phase` pins that.

The `stopReadiness` precondition holds: the guard immediately above returns early when it is absent, and `stopApplication` deletes it only after awaiting it — so by the time it is gone, the early return is the correct answer anyway.

Growth approval target moves to this head: +3,268 B, +5.09%, 7,508 B headroom.

**Comment by paulfalgout** [comment]
Claude exact-head review for `389ef16c2744c08cc19dead8f044cf1826f3130a`: **APPROVE — no blocking issues**.

Claude reviewed the complete current Application lifecycle implementation after the Cubic stop-target change and traced all four public entry points through operation creation, supersession, current-operation guards, shared stop readiness, centralized failure-state recovery, and terminal destroy behavior. It found no path that corrupts lifecycle state, double-settles an operation, repeats stop hooks, or resolves a Promise contrary to the lifecycle target reached.

This head caches one `stopPromise` derived from the active destroy operation's stop-readiness Promise. Repeated `stop()` calls during destroy therefore share referential identity, settle when stopping succeeds, reject when stopping fails, and remain independent of later `before:destroy` or `destroy` failures. Local validation: 36 focused lifecycle tests; 1,776 full tests at 100% statements, branches, functions, and lines; lint, generated distribution checks, source validation, packed fixtures, performance contracts, size, and timing passed across the reviewed change sequence. GitHub CI, final Cubic, exact-head growth approval, mergeability, and unresolved threads remain independent gates.

File: modules/application.js [hunk 3]
```diff
@@ -28,17 +36,218 @@ const Application = function(options) {
...
+  stop(options) {
+    if (this._lifecycleState === DESTROYED) {
+      return Promise.resolve(true);
+    }
+
+    const operation = this._lifecycleOperation;
+    if (this._lifecycleState === DESTROYING) {
+      if (!operation?.stopReadiness) { return Promise.resolve(true); }
+
+      return operation.promise.catch(error => {
```
[the remark this answers is not in the document]
Comment (paulfalgout) [reply modules/application.js]: Handled by caching one stop-target Promise derived from stopReadiness; repeated stop calls during destroy now share identity and settle independently from later destroy work.

**Comment by paulfalgout** [comment]
```json
{
  "schemaVersion": 1,
  "headSha": "b12e318a40707d724602c9b50be70b78dc8aa6de",
  "issueUrl": "https://github.com/marionettejs/marionette/issues/127",
  "approvedPaths": [
    "dist/marionette.cjs",
    "dist/marionette.js",
    "dist/marionette.min.js",
    "dist/marionette.umd.js"
  ],
  "evidenceUrls": [
    "https://github.com/marionettejs/marionette/issues/127#issuecomment-5476431943"
  ]
}
```

**Comment by paulfalgout** [comment]
**Both blockers I raised are cleared** at `389ef16c` — the head I reviewed and approved, unchanged. Every check now passes, including the two that were red:

- **Node 24** — the 100% coverage gate is satisfied, so `application.js:134` and `:189` are closed.
- **Bundle size** — the exact-head growth approval landed.

My approval stands as posted, and the code side is done from my end.

## One thing green CI does not cover

#326 merged first, so master's `ROADMAP.md` now says:

> Application lifecycle concurrency is a separate contract decision, not an implementation detail. ... Application alone does not establish async lifecycle by momentum.

That gate landing ahead of this PR is the good ordering — but it means merging here without recording the decision is precisely the momentum the clause names. The gate also now asks that the chosen contract *"define which hooks and notifications are awaited, their failure semantics, and their overlap behavior."*

Worth restating what that costs, because it is less than it sounds. The implementation already answers all three:

| Requirement | Status in this PR |
| --- | --- |
| which hooks are awaited | `before:start`, `before:stop`, `before:destroy` awaited; `start`, `stop`, `destroy` synchronous |
| failure semantics | `getFailureState`, `stopReadiness` inheritance, the `completing` / `stopped` split |
| overlap behavior | supersession, shared readiness, reentry, repeated cycles — all in the lifecycle spec |

So the outstanding work is writing those down as a decision plus the evidence half — verified public consumer races, migration cost, agent discoverability — not rebuilding anything. Whether that belongs in this PR or a follow-up is a maintainer call; flagging it only so the sequencing is deliberate rather than incidental.

File: modules/application.js [hunk 4]
```diff
@@ -28,17 +36,269 @@ const Application = function(options) {
...
+      // Destroy cannot be superseded and starts its stop phase synchronously.
+      if (!operation.stopResult) {
+        operation.stopResult = createDeferred();
+      }
+      return operation.stopResult.promise;
+    }
+    if (operation?.kind === 'stop') { return operation.promise; }
+    if (operation?.isStopped) {
+      const superseded = supersedeOperation(this);
+      superseded.readiness?.controller.abort();
```
[the remark this answers is not in the document]
Comment (paulfalgout) [reply modules/application.js]: Fixed in 1ac46385. The already-stopped shortcut now publishes STOPPED before abort dispatch, so a reentrant destroy observes the completed stop phase and does not run stop teardown again. A regression test asserts one before:stop/stop pair and successful reentrant destroy.

File: modules/application.js [hunk 5]
```diff
@@ -28,17 +36,269 @@ const Application = function(options) {
...
+  })();
+}
+
+function beginOperation(application, kind, state, failureState, callback) {
+  const superseded = supersedeOperation(application);
+  const deferred = createDeferred();
+  const stopReadiness = superseded?.stopReadiness;
+
+  if (superseded?.readiness && superseded.readiness !== stopReadiness) {
+    superseded.readiness.controller.abort();
```
[the remark this answers is not in the document]
Comment (paulfalgout) [reply modules/application.js]: Fixed in 1ac46385. beginOperation now installs the replacement operation and transition state before abort dispatch, then verifies it is still current before starting readiness. A reentrant abort listener may supersede it without being overwritten; the invalidated replacement resolves false and the winning destroy settles.

File: docs/marionette.application.md [hunk 6]
```diff
@@ -34,12 +33,71 @@ import { Application } from 'marionette';
...
+The `onBeforeStart`, `onBeforeStop`, and `onBeforeDestroy` methods may return a
+Promise. Their corresponding `before:*` events still fire synchronously, but
+event-listener return values are not readiness inputs. `onStart`, `onStop`,
+`onDestroy`, and their matching events are completion notifications and are not
+awaited. A `before:*` method must not await the same operation whose readiness it
+is defining. `restart` composes the stop and start lifecycles; it does not add a
+parallel restart hook path.
+
+Each readiness method and `before:*` event receives the Application, the
+operation options, and a context object with an [`AbortSignal`](https://developer.mozilla.org/en-US/docs/Web/API/AbortSignal):
```
[the remark this answers is not in the document]
Comment (paulfalgout) [reply docs/marionette.application.md:85]: The premise is superseded by the selected roadmap contract merged in #330 and incorporated here at f0093ba8. ROADMAP.md now explicitly selects a platform AbortSignal for Application readiness and distinguishes that bounded hook argument from adding a general signals runtime. This PR implements that selected v5 contract rather than a prohibited v6/labs surface.

File: docs/events.class.md [hunk 7]
```diff
@@ -81,17 +83,35 @@ const MyApp = Application.extend({
...
 
-As shown the `options` object is passed into the `Application` as the
-second argument to `start`.
+As shown, the `options` object passed to `start` is forwarded after the
+Application to its lifecycle methods and events. Readiness methods and
+`before:*` events also receive a context object as the third argument. Its
+`signal` is aborted when a later operation invalidates that readiness. A
+transferred stop phase retains the same context and un-aborted signal. Only the
+Promise returned by the `onBeforeStart`, `onBeforeStop`, or `onBeforeDestroy`
+method delays its lifecycle; event-listener return values are ignored.
```
[the remark this answers is not in the document]
Comment (paulfalgout) [reply docs/events.class.md]: Handled in f6e9c775. I verified all three linked Application JSFiddles still load the v4-era backbone.marionette/next build and call start without await. Because this task is repo-only and does not authorize publishing external examples, the stale links were removed while the complete v5 inline examples remain.
